File: integrations/aspnetcore/src/Scalar.AspNetCore/Options/Authentication/Flows/ImplicitFlow.cs
using System.Text.Json.Serialization;

namespace Scalar.AspNetCore;

/// <summary>
/// Represents the OAuth2 Implicit flow configuration.
/// </summary>
public sealed class ImplicitFlow : OAuthFlow
{
    /// <summary>
    /// Gets or sets the authorization URL to be used for this flow.
    /// </summary>
    public string? AuthorizationUrl { get; set; }

    /// <summary>
    /// Gets or sets the redirect URI for the implicit flow.
    /// </summary>
    [JsonPropertyName("x-scalar-redirect-uri")]
    public string? RedirectUri { get; set; }
}

File: integrations/java/webjar/src/main/java/com/scalar/maven/webjar/authentication/flows/ImplicitFlow.java
package com.scalar.maven.webjar.authentication.flows;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Represents the OAuth2 Implicit flow configuration.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ImplicitFlow extends OAuthFlow {

    /**
     * Gets or sets the authorization URL to be used for this flow.
     */
    private String authorizationUrl;

    /**
     * Gets or sets the redirect URI for the implicit flow.
     */
    @JsonProperty("x-scalar-redirect-uri")
    private String redirectUri;

    /**
     * Creates a new ImplicitFlow.
     */
    public ImplicitFlow() {
    }

    /**
     * Gets the authorization URL.
     *
     * @return the authorization URL
     */
    public String getAuthorizationUrl() {
        return authorizationUrl;
    }

    /**
     * Sets the authorization URL.
     *
     * @param authorizationUrl the authorization URL
     */
    public void setAuthorizationUrl(String authorizationUrl) {
        this.authorizationUrl = authorizationUrl;
    }

    /**
     * Gets the redirect URI.
     *
     * @return the redirect URI
     */
    public String getRedirectUri() {
        return redirectUri;
    }

    /**
     * Sets the redirect URI.
     *
     * @param redirectUri the redirect URI
     */
    public void setRedirectUri(String redirectUri) {
        this.redirectUri = redirectUri;
    }
}
